piptools/resolver.py

```diff
@@ -104,10 +117,29 @@ def combine_install_requirements(
     return combined_ireq
 
 
-class Resolver:
+class BaseResolver:
+    repository: BaseRepository
+
+    def resolve(self, max_rounds: int) -> Set[InstallRequirement]:
+        raise NotImplementedError("Override in subclass")
+
+    def resolve_hashes(
+        self, ireqs: Set[InstallRequirement]
+    ) -> Dict[InstallRequirement, Set[str]]:
+        """
+        Finds acceptable hashes for all of the given InstallRequirements.
+        """
```

Comment:
Methods/functions perform actions so the docstrings should start with a verb and be in imperative mood as opposed to objects/classes/modules/packages where the docstrings merely describe what they are.
```suggestion
        """Find acceptable hashes for all of the given ``InstallRequirement``s."""
```

Reply:
Addressed in 81419f9f1090bda0761020e7e9832ba279a5fd4d